broker/src/main/java/io/camunda/zeebe/broker/transport/partitionapi/InterPartitionCommandReceiverImpl.java

```diff
@@ -141,10 +144,13 @@ DecodedMessage decodeMessage(final byte[] message) {
       // are a valid command
       final var commandOffset =
           messageDecoder.limit() + InterPartitionMessageDecoder.commandHeaderLength();
-      final var commandLength = messageDecoder.commandLength();
-      commandBuffer.wrap(messageBuffer, commandOffset, commandLength);
 
-      return new DecodedMessage(checkpointId, recordKey, recordMetadata, commandBuffer);
+      final var commandLength = messageDecoder.commandLength();
+      final var value =
+          ReflectUtil.newInstance(
```

Comment:
:question: What happens when we receive an unknown instance? An exception is thrown? If so, is it descriptive enough for us to understand quickly what's happening? I guess we want it to fail here rather than further downstream.

Reply:
I'll explicitly log this error. Since there are plans to change resource deployment there's a decent chance that this happens during rolling updates.